altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -112,110 +135,227 @@ pub fn start_scheduler() {
+        assert_eq!(handle_3.tid(), Ok(test::current_task().unwrap().tid()));
     
-    switch_context();
-    assert!(test::current_task().is_some());
-    assert_eq!(handle_3.tid(), Ok(test::current_task().unwrap().tid()));
-  }
+        switch_context();
+        assert!(test::current_task().is_some());
+        assert_eq!(handle_3.tid(), Ok(test::current_task().unwrap().tid()));
+    }
+
+    #[test]
+    //Scheduler runs with only Critical and no Normal or Low Priority Tasks.
+    //TODO: Refactor the next 3 tests into 1...
+    fn test_scheduler_runs_with_only_critical_tasks() {
+        let _g = test::set_up();
+        test::create_and_schedule_test_task(512, Priority::Critical, "test task 1");
+        test::create_and_schedule_test_task(512, Priority::Critical, "test task 2");
+        start_scheduler();
+        for _ in 0..100 {
+            assert!(test::current_task().unwrap().priority == Priority::Critical);
+        }
+    }
+
+    #[test]
+    //Scheduler runs with only Normal and no Critical or Low Priority Tasks.
+    fn test_scheduler_runs_with_only_normal_tasks() {
+        let _g = test::set_up();
+        test::create_and_schedule_test_task(512, Priority::Normal, "test task 1");
+        test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
+        start_scheduler();
+        for _ in 0..100 {
+            assert!(test::current_task().unwrap().priority == Priority::Normal);
+        }
+    }
+
+    #[test]
+    //Scheduler runs with only Low and no Normal or Critical Priority Tasks.
+    fn test_scheduler_runs_with_only_low_tasks() {
+        let _g = test::set_up();
+        test::create_and_schedule_test_task(512, Priority::Low, "test task 1");
+        test::create_and_schedule_test_task(512, Priority::Low, "test task 2");
+        start_scheduler();
+        for _ in 0..100 {
+            assert!(test::current_task().unwrap().priority == Priority::Low);
+        }
+    }
+
+    #[test]
+    //Scheduler selects __Idle task if run when no tasks queued.
+    fn test_pick_idle_when_no_task_in_queues() {
+        let _g = test::set_up();
+        start_scheduler();
+        assert_eq!(test::current_task().unwrap().priority, Priority::__Idle);
+    }
+
+    #[test]
+    //Scheduler selects __Idle task if run when all taks are blocked.
+    fn test_pick_idle_when_all_tasks_are_blocked() {
+        let _g = test::set_up();
+        assert!(test::current_task().is_none());
+        let handle_1 = test::create_and_schedule_test_task(512, Priority::Low, "test task 1");
+        let handle_2 = test::create_and_schedule_test_task(512, Priority::Normal, "test task 2");
+        let handle_3 = test::create_and_schedule_test_task(512, Priority::Critical, "test task 3");
+        start_scheduler();
+        assert!(test::current_task().is_some());
+        test::current_task().unwrap().state = State::Blocked;
+        test::current_task().unwrap().delay_type = Delay::Sleep;
```

Comment:
It should be a very quick fix, doesn't seem like its worth the effort to make an issue out of it and wait to fix it when we could just get it done now.